src/Recommendations.jsx

```diff
@@ -30,6 +31,33 @@ function Recommendations() {
         setRecommendations(data);
       });
   }
+  useEffect(() => {
+    if (recommendations) {
+      const fetchAllMovieDetails = async () => {
+        const movieCards = await Promise.all(
+          recommendations.map(async (movie) => {
+            const response = await fetch(
+              `http://localhost:3000/movies/${movie.itemID}`
+            );
+            if (!response.ok) {
+              throw new Error("Failed to fetch movie details");
+            }
+            const details = await response.json();
+            return (
+              <MovieCard
+                key={movie.itemID}
+                movieRating={movie.rating * 2}
+                movieTitle={details.title}
+              />
+            );
+          })
+        );
+        setMovieCards2(movieCards.filter((card) => card !== null));
```

Comment:
It's better not to store markup in state. The better practice is to store the data in object form in the state. 
Then parse the state objects (with `Array.map` if you will) to produce markup in the JSX part.